iree/compiler/Dialect/HAL/Target/LLVM/LLVMTargetOptions.cpp

```diff
@@ -158,6 +158,15 @@ LLVMTargetOptions getLLVMTargetOptionsFromFlags() {
       llvm::cl::init(llvmTargetOptions.keepLinkerArtifacts));
   llvmTargetOptions.keepLinkerArtifacts = clKeepLinkerArtifacts;
 
+  static llvm::cl::opt<std::string> clStaticLibraryOutputPath(
+      "iree-llvm-static-library-output-path",
+      llvm::cl::desc(
+          "Path for static library output (EX: \"path/to/static-library\"). "
+          "This will produce the static library at the specified path along "
+          "a similarly named \".h\" file for static linking."),
+      llvm::cl::init(llvmTargetOptions.staticLibraryOutput));
```

Comment:
Updated based on suggestion. The '.o' or '.h' in the library name is optional, since the code specifically adds those suffixes, but specifying one does make the flag use clearer. 

LLVM actually generates object files with '.obj' but I rename them to '.o' since that is the convention when linking c++ programs (pretty much any example I've seen).